aspnet/fundamentals/file-providers/sample/src/FileProviderSample/Controllers/HomeController.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.FileProviders;
+
+namespace FileProviderSample.Controllers
+{
+    public class HomeController : Controller
+    {
+        private readonly IFileProvider _fileProvider;
+        private static string _dateFileLastChanged = "Never";
+
+        public HomeController(IFileProvider fileProvider)
+        {
+            _fileProvider = fileProvider;
+        }
+
+        public IActionResult Index()
+        {
+            var contents = _fileProvider.GetDirectoryContents("");
+            return View(contents);
+        }
+
+        public IActionResult Watch()
```

Comment:
Ok, I switched to use your sample, thanks. I like that it has all of the token/registerchangecallback code in one place (though it is more complicated to follow, overall, with the async and TCS).